andela/eagle-bn-backend
# #168781794 Setup Eslint to follow airbnb style guide

#### What does this PR do?
- setup Eslint to follow Airbnb style guide
#### Description of Task to be completed?
- setup Eslint
- configure Eslint with Airbnb style guides
-  install Eslint packages in dev-dependences packages include: `Eslint`, `eslint-config-airbnb-base`, `eslint-plugin-import`.
- add new rules in eslint configuration as follows
```
 "root": true,
  "extends": "airbnb-base",
  "env": {
    "node": true,
    "es6": true,
    "mocha": true
  },
  "rules": {
    "one-var": 0,
    "one-var-declaration-per-line": 0,
    "new-cap": 0,
    "consistent-return": 0,
    "no-param-reassign": 0,
    "comma-dangle": 0,
    "curly": ["error", "multi-line"],
    "import/no-unresolved": [2, { commonjs: true }],
    "no-shadow": ["error", { "allow": ["req", "res", "err"] }],
    "valid-jsdoc": ["error", {
      "requireReturn": true,
      "requireReturnType": true,
      "requireParamDescription": false,
      "requireReturnDescription": true
    }],
    "require-jsdoc": ["error", {
        "require": {
            "FunctionDeclaration": true,
            "MethodDefinition": true,
            "ClassDeclaration": true
        }
    }]
  }
}

```
#### How should this be manually tested?
- clone the repo
- run npm i  to install the packages
- write some code to see the style guides
#### Any background context you want to provide?
nan
#### What are the relevant pivotal tracker 

[ #168781794](https://www.pivotaltracker.com/story/show/168781794l)

#### Screenshots (if appropriate)
nan
#### Questions:
nan
